src/Services/Air/Air.js

```diff
@@ -269,8 +296,9 @@ module.exports = (settings) => {
         }
 
         const pnr = await this.getPNRByTicketNumber({ ticketNumber });
-        const tickets = await this.getTickets({ pnr });
-        const ticket = tickets.find(t => t.ticketNumber === ticketNumber);
+        const ticket = err instanceof AirRuntimeError.DuplicateTicketFound
```

Comment:
Hard to expand this kind of handling to the third error.

```
const { helpers: { getFullName: getFullErrorName } } = require('../../node-errors-helpers');

retryableTicketErrorHandlers = {
  'AirRuntimeError.TicketInfoIncomplete': async (ticketNumber) => {
    const pnr = await this.getPNRByTicketNumber({ ticketNumber });
    const tickets = await this.getTickets({ pnr });
    return tickets.find(t => t.ticketNumber === ticketNumber);
  },
  'AirRuntimeError.DuplicateTicketFound': async (ticketNumber) => {
    const pnr = await this.getPNRByTicketNumber({ ticketNumber });
    const { splitBookings } = await this.getBooking({ pnr });
    if (!splitBookings) {
      return null;
    }
    const [splitBookingPNR] = splitBookings;
    const { uapi_ur_locator: urLocator } = await this.getBooking({ pnr: splitBookingPNR });
    return service.getTicket({
      ticketNumber,
      uapi_ur_locator: urLocator,
      pnr: splitBookingPNR,
    });
  },
}

async getTicket(options) {
  const { ticketNumber, allowNoProviderLocatorCodeRetrieval = false } = options;
  try {
    return await service.getTicket({ ticketNumber, allowNoProviderLocatorCodeRetrieval });
  } catch (err) {
    const errorName = getFullErrorName(err);
    const retryableErrorHandler = this.retryableTicketErrorHandlers[errorName];
    const ticket = await(retryableErrorHandler && retryableErrorHandler(ticketNumber));
    if (!ticket) {
      throw err;
    }
    return ticket
  }
}
```

Please use the approach similar to the described above, where you can register a handler for each error type